### DISCUSSION TITLE
How to remove a value from the `Extension.Store` of another hierarchy

### DISCUSSION THREAD

**Discussion by bjmi** [junit-team/junit-framework#3757] [General]
Given an extension that implements `InvocationInterceptor` and `AfterEachCallback`.
```
public <T> T interceptTestClassConstructor(Invocation<T> invocation,
                                           ReflectiveInvocationContext<Constructor<T>> invocationContext,
                                           ExtensionContext extensionContext) throws Throwable {
    var store = extensionContext.getStore(MY_NAMESPACE);
    store.put("mykey", "myvalue");
    return invocation.proceed();
}
public void afterEach(ExtensionContext extensionContext) {
    var store = extensionContext.getStore(MY_NAMESPACE);
    var removed = store.remove("mykey") // removed is null
    var value = store.get("mykey")      // value is "myvalue"
}
```
The store in _interceptTestClassConstructor(...)_ stems from a `ClassExtensionContext` and the store in _afterEach(...)_ from a `MethodExtensionContext`.

What would be a proper solution to remove the value from the store in _afterEach(...)_?
(BeforeEachCallback can't be used as some statements are already executed in the constructor of the test instance.)

**Comment by marcphilipp** [comment]
You can access the parent `ExtensionContext` via `ExtensionContext.getParent()` and, if it's present, its store. If required, you can apply this until you reach the root context.
